apsuite/commissioning_scripts/meas_coupling_tune.py

```diff
@@ -17,201 +41,203 @@ class FitTunes(_SimulAnneal):
-            eigvalues, _ = _np.linalg.eig(mat)
-            tune1[i], tune2[i] = eigvalues
-            if tune1[i] <= tune2[i]:
-                tune1[i], tune2[i] = tune2[i], tune1[i]
-        return param, tune1, tune2
-
-    def fitting_plot(
-            self, oversampling=1, title=None, xlabel=None, fig=None):
-        """."""
-        position = self.position
-        param = _np.interp(
-            _np.arange(0, len(self._param), 1/oversampling),
-            _np.arange(0, len(self._param)),
-            self._param)
-
-        # NOTE: do error estimate properly!
-        coup_error = self.calc_obj_fun()
+    def ismeasuring(self):
+        """."""
+        return self._thread.is_alive()
+
+    def _do_meas(self):
+        quad = self.devices['quad']
+        tunes = self.devices['tune']
+
+        curr0 = quad.current
+        curr_vec = curr0 * _np.linspace(
+            1-self.params.neg_percent,
+            1+self.params.pos_percent,
+            self.params.nr_points)
+        tunes_vec = _np.zeros((len(curr_vec), 2))
+
+        print('QFB Current:')
+        for idx, curr in enumerate(curr_vec):
+            quad.current = curr
+            _time.sleep(self.params.time_wait)
+            tunes_vec[idx, :] = tunes.tunex, tunes.tuney
+            print('   {:.6f} A ({:+.3f} %), nux = {:.4f}, nuy = {:.4f}'.format(
+                curr, (curr/curr0-1)*100),
+                tunes_vec[idx, 0], tunes_vec[idx, 1])
+        print('Finished!')
+        quad.current = curr0
+        self.data['qname'] = quad.devname
+        self.data['current'] = curr_vec
+        self.data['tunes'] = tunes_vec
+
+    def process_data(self, coup_resolution=None):
+        """."""
+        qcurr, tune1, tune2 = self._filter_data(coup_resolution)
+        ini_param = self._calc_init_parms(qcurr, tune1, tune2)
+        # least squares using Levenberg-Marquardt minimization algorithm
+        fit_param = least_squares(
+            fun=MeasCoupling._err_func, x0=ini_param,
+            args=(qcurr, tune1, tune2), method='lm')
+
+        self.analysis['qcurr'] = qcurr
+        self.analysis['tune1'] = tune1
+        self.analysis['tune2'] = tune2
+        self.analysis['initial_param'] = ini_param
+        self.analysis['fitted_param'] = fit_param
+        fit_error = self._calc_fitting_error()
+        self.analysis['fitting_error'] = fit_error
+
+    def plot_fitting(
+            self, oversampling=1, title=None, xlabel=None, fig=None,
+            save=False, fname=None):
+        """."""
+        anl = self.analysis
+        fit_vec = anl['fitted_param']['x']
+        qcurr, tune1, tune2 = anl['qcurr'], anl['tune1'], anl['tune2']
+
+        qcurr_interp = _np.interp(
+            _np.arange(0, len(qcurr), 1/oversampling),
+            _np.arange(0, len(qcurr)), qcurr)
+
+        fittune1, fittune2 = MeasCoupling._get_normal_modes(
+            params=fit_vec, curr=qcurr_interp)
 
+        if fig is None:
```

Comment:
I think the optional argument should be the `axes` object, not the figure. I recommend substitute the `_plt.` by the `axes.`